src/applications/check-in/components/UpcomingAppointmentsList.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+// eslint-disable-next-line import/no-unresolved
+import { recordEvent } from '@department-of-veterans-affairs/platform-monitoring/exports';
+import { createAnalyticsSlug } from '../utils/analytics';
+import { useFormRouting } from '../hooks/useFormRouting';
+
+import {
+  getAppointmentId,
+  organizeAppointmentsByYearMonthDay,
+} from '../utils/appointment';
+
+import UpcomingAppointmentsListItem from './UpcomingAppointmentsListItem';
+
+import { makeSelectVeteranData } from '../selectors';
+
+const UpcomingAppointmentsList = props => {
+  const { router, app } = props;
+  const { jumpToPage } = useFormRouting(router);
+  const selectVeteranData = useMemo(makeSelectVeteranData, []);
+  const { upcomingAppointments } = useSelector(selectVeteranData);
+
+  const groupedAppointments = organizeAppointmentsByYearMonthDay(
+    upcomingAppointments,
+  );
+
+  const handleDetailClick = (e, appointment) => {
+    e.preventDefault();
+    recordEvent({
+      event: createAnalyticsSlug('details-link-clicked', 'nav', app),
+    });
+    jumpToPage(`appointment-details/${getAppointmentId(appointment)}`);
+  };
+
+  if (groupedAppointments.length < 1) {
+    window.scrollTo(0, 0);
+    return (
+      <div data-testid="no-upcoming-appointments">
+        You have no upcoming appointments
+      </div>
+    );
+  }
+  return (
+    <>
+      {groupedAppointments.map(month => {
+        const { monthYearKey, days } = month;
+
+        return (
+          <div key={monthYearKey}>
+            <h3 data-testid="appointments-list-monthyear-heading">
+              {`${monthYearKey.split('-')[1]} ${monthYearKey.split('-')[0]}`}
+            </h3>
+            <ul
+              className="vads-u-margin-bottom--4 check-in--appointment-list appointment-list"
+              data-testid="appointment-list"
+            >
+              {days.map(day => {
+                const { appointments, dayKey } = day;
+                return (
+                  <>
```

Comment:
these children need a unique key. you can use <React.Fragment key={uniqueKey}> to do that